lib/internal/main/repl.js

```diff
@@ -11,8 +11,15 @@ const {
   evalScript
 } = require('internal/process/execution');
 
+const { ERR_INVALID_REPL_TYPE } = require('internal/errors').codes;
+
 prepareMainThreadExecution();
 
+// --entry-type flag not supported in REPL
+if (require('internal/process/esm_loader').typeFlag) {
```

Comment:
We could either exit the process early in `StartMainThreadExecution()` before even calling into JS land, or do this early here before `prepareMainThreadExecution()` with `getOptionValue('--entry-type')`.

Reply:
What is your concern with this one?

I wasn't sure what the error convention would be if we move this into `StartMainThreadExecution`, as there aren't any other errors thrown there.

This seems most similar to the existing eval + breakEvalOnSigInt sort of validation to me, which actually happens higher up in the REPL JS side itself, not lower.

Reply:
Anything we still need to change here?

Reply:
It may involve more changes if you want to change `StartMainThreadExecution` (e.g. you need to stop using `LoadEnvironment` in `StartNodeWithIsolate` which is fine since  `LoadEnvironment` does not really work for embedders anyway).

The easiest way to do this is:

```
if (getOptionValue('--entry-type')) {
  console.error('Cannot specify --entry-type for REPL');
  process.exit(1);
}
```
, we use to do that for `node --debug` before moving it to end-of-life.

Reply:
Sure, I've pushed up a commit with that change!